tensorboard/uploader/auth.py

```diff
@@ -149,36 +174,188 @@ def clear(self):
                 raise
 
 
-def build_installed_app_flow(client_config):
-    """Returns a `CustomInstalledAppFlow` for the given config.
+def authenticate_user(
+    force_console=False,
+) -> google.oauth2.credentials.Credentials:
+    """Makes the user authenticate to retrieve auth credentials."""
+    flow = _TbUploaderAuthFlow(
+        _BROWSER_OAUTH_CLIENT_CONFIG,
+        _CONSOLE_OAUTH_CLIENT_CONFIG,
+        OPENID_CONNECT_SCOPES,
+    )
+    return flow.run(force_console)
 
-    Args:
-      client_config (Mapping[str, Any]): The client configuration in the Google
-          client secrets format.
 
-    Returns:
-      CustomInstalledAppFlow: the constructed flow.
-    """
-    return CustomInstalledAppFlow.from_client_config(
-        client_config, scopes=OPENID_CONNECT_SCOPES
-    )
+class _TbUploaderAuthFlow(auth_flows.Flow):
```

Comment:
Ended up removing the base class. I think it was misleading and erroneous, because the base methods don't really make sense (or work, probably) for this flow.